intro-to-functions/defining-functions.md

```diff
@@ -0,0 +1,455 @@
+# Defining Functions
+
+## Introduction
+
+A function is a reusable chunk of code. Every function is defined with a name and some logic.
+
+If, in a given project, we want to **define** our own reusable chunks of code, then we can define our own functions. Then, any code that has access to our functions can invoke them. To define a function, we must follow specific Python syntax.
+
+## Learning Goals
+
+- Apply Single Responsibility Principle to functions
+- Define a function (function signature, function body, return)
+- Demonstrate calling functions with and without arguments/parameters
+
+## Vocabulary and Synonyms
+
+| Vocab | Definition | Synonyms | How to Use in a Sentence
+| --- | --- | --- | ---
+| Function | Lines of code (1 or more) that are related, grouped together, and named. Once defined, these lines of code are reusable and can be called over and over again. | Method | "I defined a function," "I used a function," "I wrote a function"
+| Function definition, defining a function | How a function gets defined before it gets invoked | Function def, where the function is defined, a function you wrote | "I defined a function named `get_length`," "I defined the `get_length` function in this file," "I had bugs in my `get_length` function definition."
+| Function Signature | A piece of syntax. A part of the function definition that determines function name and the parameter list | Method signature | "We can read the name from the function signature," "The function signature tells us what parameters there are."
+| Parameter | The name of an expected argument for this function. This parameter name is the name of the local variable that will hold an argument value. Parameters and arguments at the time of function call are mapped positionally. | "Argument." Param. What is passed into a function. | "This function has one parameter named `items`," "There are 3 parameters for this function," "The function doesn't take any parameters"
+
+## Every Function Has a Responsibility
+
+Remember, functions do things, and have a responsibility to do certain things when they are called.
+
+Before we create a function, we should do our best to determine:
+
+1. What is the kind of logic you want to put into a function? What is this function _responsible for doing_?
+1. What is the best name for this function? Ideally, the name of the function implies the responsibility of this function.
+1. What kind of arguments should the function take in? How many arguments?
+1. What one piece of data should the function return? What is its data type, and what does it represent?
```

Comment:
```suggestion
1. What is the kind of logic you want to put into a function?
    - What is this function _responsible for doing_?
1. What is the best name for this function? 
    - Ideally, the name of the function implies the responsibility of this function.
1. What kind of arguments should the function take in? 
    - How many arguments?
1. What one piece of data should the function return? 
    - What is its data type, and what does it represent?
```

I feel really confident that GH markdown needs four-space indent to use mixed lists

Push back if you think this isn't the case, otherwise I'll merge this at some point